### DISCUSSION TITLE
Additional Lisps

### DISCUSSION THREAD

**Discussion by mikelevins** [rabbibotton/clog#85] [General]
What's your level of interest in extending support for additional Lisps? I might possibly have a use for Clog in a project built on Allegro CL.

**Comment by rabbibotton** [comment]
I very much would like. I test on ecl and sbcl currently. If I had access
and time I would make sure ran on those as well. So any help would be
appreciated.

**Reply by mikelevins** [reply]
I'd be happy to help, but my Allegro license actually belongs to a client, so I need to justify its use appropriately.

Clog looks like a decent candidate for developing some GUIs that we presently handle in other ways. If I can ensure that Clog works on Allegro and saves me as much time as I expect it to, then I can make a case for offering some testing support and maybe some help with Allegro-specific issues. The chief roadblocks are (1) getting it working on Allegro and (2) licensing: Clack is licensed LLGPL, and that's not a whitelisted license for my client.

If surmounting these obstacles is within what you regard as reasonable effort, I'd be glad to help. If not, I completely understand.

**Comment by rabbibotton** [comment]
>
>
>  (1) getting it working on Allegro
>
 I don't imagine an issue as it is a very clean basic CL, nothing specific
to any implementation.


> and (2) licensing: Clack is licensed LLGPL, and that's not a whitelisted
> license for my client.
>
clog-connection is the only file you would need to touch to switch out to
hard wire to a specific http server or perhaps could ask the owner of Clack
to consider a less restrictive license to start with (and likely should
have considering his intent for the project.)


>

**Reply by mikelevins** [reply]
I did try just naively loading clog into Allegro 10.1 on Windows 10 and ran into issues right away. One was the problem of finding sqlite (which I reported in another Windows-related discussion thread, and which I can solve the same way--by sticking a copy of the DLL in the directory with the Allegro executable--that seems slightly gross to me, but the other ways to do it are significantly more of a pain).

Another problem is that the CFFI groveler for static-vectors (required by websocket-driver) chokes on processing ffi-types.lisp. I didn't investigate further, but I suspect that CFFI is expecting to find a C compiler and failing (I'm running it on Windows 10 with Visual Studio 2019 installed with the C++ workload, so the compiler is there; just presumably not where CFFI expects to find it). Most likely, this problem is an easy configuration issue, but will require some groveling around (heh) to figure out what config to change.

I naively expect to run into more things like this for the simple reason that I've been doing this sort of thing for thirty-some-odd years and I _always_ run into stuff like this.

**Comment by rabbibotton** [comment]
What is the allegro’s solution for http. Just replace clack with that and
solved for this as well.

**Reply by mikelevins** [reply]
I imagine I can hack in hunchentoot and hunchensocket. If clog-connection is the only file affected, that might do the trick (thanks for that guidance). I’m too old and battle scarred to expect that to solve everything, but if it does, how wonderful!

I’ll look into it.

**Reply by mikelevins** [reply]
Oh, or allegroserve, which is built in and offers web socket support.
